packages/orchestrator-ui-components/src/components/WFOSteps/WFOStepList/WFOStepList.tsx

```diff
@@ -104,6 +131,32 @@ export const WFOStepList = ({ steps = [], startedAt }: WFOStepListProps) => {
             </EuiFlexGroup>
             <>
                 {steps.map((step, index) => {
+                    let prevState = {};
+                    let delta = {};
+
+                    if (index > 0) {
+                        // Todo: decided if we want this here.
+                        //  V1 has a rather big case statement with businness logic here
+                        if (steps[index].status === 'success') {
+                            // Prepare a delta with the last successful step
+                            let prev_index = index - 1;
+                            while (
+                                prev_index > 0 &&
+                                (steps[prev_index].status === 'failed' ||
```

Comment:
To make this function a bit more pure: you could use the 3rd argument from the mapper function that is the whole array too